plugins/modules/greppy.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/python
+
+# SPDX-FileCopyrightText: 2022, David Peng <dpeng1@gmail.com>
+# SPDX-License-Identifier: GPL-3.0-or-later
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: greppy
+
+short_description: Scan logs for certain patterns within a given timeframe.
+
+# If this is part of a collection, you need to use semantic versioning,
+# i.e. the version is of the form "2.5.0" and not "2.4".
+
+description: Grep logs for a list of patterns within a given timeframe.
+
+options:
+    path:
+        description: The file to grep.
+        required: true
+        type: path
+    search:
+        description: The literal string to look for in every line of the file. This does not have to match the entire line.
+        required: true
+        type: list
+        elements: str
+    exclude:
+        description: The literal string to exclude from the file. This does not have to match the entire line.
+        required: false
+        type: list
+        elements: str
+    timeout:
+        description: The number of seconds to wait.
+        required: false
+        default: 60
+        type: int
+    ignore_case:
+        description: Case insensitive regex search.
+        required: false
+        type: bool
+    findexit:
+        description: Exit right after finding the first occurrence
+        required: false
+        default: false
+        type: bool
+# Specify this value according to your collection
+# in format of namespace.collection.doc_fragment_name
+extends_documentation_fragment:
+    - community.general.greppy
+
+author:
+    - David Peng (@dpengftw)
+'''
+
+EXAMPLES = '''
+- name: Grep for strings in given list but also exclude certain strings
+  community.general.greppy:
+    path: example.txt
+    search:
+      - first string
+      - second string
+    exclude:
+      - third
+      - fourth
+
+- name: Grep for strings and wait for 120 seconds
+  community.general.greppy:
+    path: example.txt
+    search:
+      - first string
+    timeout: 120
+
+- name: Grep for first occurrence then exit
+  community.general.greppy:
+    path: example.txt
+    search:
+      - first occurrence
+    find_exit: yes

```

Comment:
```suggestion
    find_exit: true
```
Always use `true` and `false` for YAML booleans. Never use `True`, `yes`, ...